### PR TITLE
Fixes #1605 #1668 #1678 #1692  #133

### LINKED ISSUE #133: Leetcode Question Number 143
Leetcode Question Number 143 soultion needs to be added in the repo . 
 The link for the question is [this](https://leetcode.com/problems/reorder-list).
 Accepted Languages C++ / JAVA . 
 You Only need to add the  class Solution in the file. 
 Make Sure you have read the contributions guidelines before creating an PR.

### LINKED ISSUE #1605: Leetcode Question Number 1877
Leetcode Question Number 1877 solution needs to be added in the repo . 
 The link for the question is [this](https://leetcode.com/problems/minimize-maximum-pair-sum-in-array).
 Accepted Languages C++ / JAVA . 
 You Only need to add the  class Solution in the file. 
 Make Sure you have read the contributions guidelines before creating an PR.

### LINKED ISSUE #1668: Leetcode Question Number 1952
Leetcode Question Number 1952 solution needs to be added in the repo . 
 The link for the question is [this](https://leetcode.com/problems/three-divisors).
 Accepted Languages C++ / JAVA . 
 You Only need to add the  class Solution in the file. 
 Make Sure you have read the contributions guidelines before creating an PR.

### LINKED ISSUE #1678: Leetcode Question Number 1963
Leetcode Question Number 1963 solution needs to be added in the repo . 
 The link for the question is [this](https://leetcode.com/problems/minimum-number-of-swaps-to-make-the-string-balanced).
 Accepted Languages C++ / JAVA . 
 You Only need to add the  class Solution in the file. 
 Make Sure you have read the contributions guidelines before creating an PR.

### LINKED ISSUE #1692: Leetcode Question Number 1985
Leetcode Question Number 1985 solution needs to be added in the repo . 
 The link for the question is [this](https://leetcode.com/problems/find-the-kth-largest-integer-in-the-array).
 Accepted Languages C++ / JAVA . 
 You Only need to add the  class Solution in the file. 
 Make Sure you have read the contributions guidelines before creating an PR.

### PR DESCRIPTION
**Pull request by Anuragkillswitch**
Fixes #1605  , 
Fixes #1668 , 
Fixes #1678  ,
Fixes #1692  ,
Fixes #133 

My code sample is in : 
- [ ] CPP
- [x] Java 

Screenshot of working code sample  : 

image
image

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My code sample is formatted and works correctly 

[Contributor Guide]: https://github.com/amanv8060/Leetcode-Questions/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: solutions/1985.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (amanv8060) [inline solutions/1985.java:1]: import java.math.BigInteger;

Comment (amanv8060) [reply solutions/1985.java:1]: You will explicitly need to import this class

**Review by amanv8060 — CHANGES_REQUESTED** [review]
One change needs to be done

File: solutions/1985.java (hunk 1 above)
Comment (Anuragkillswitch) [reply solutions/1985.java:3]: Done

**Pull request merged**